### PR TITLE
Add ACMG VUS Bayesian score / temperature on case reports

### LINKED ISSUE #5099: ACMG VUS Bayesian score / temperature on case reports
## Display ACMG VUS temp also on case report.

Customers (cust002) are happy with VUS temp enough to see it on reports to allow med personell to weigh VUS better.

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
- Closes #5099

image

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by DN, CR

### CODE REVIEW

File: scout/server/blueprints/cases/controllers.py [hunk 1]
```diff
@@ -663,6 +665,19 @@ def _append_evaluated_variant_by_type(
...
+                eval_category == "classified_detailed"
+            ):  # Append info to display the ACMG VUS Bayesian score / temperature
+                variant_acmg_classifications = store.get_evaluations_case_specific(
+                    document_id=var_obj["_id"]
+                )
+                if variant_acmg_classifications:
+                    var_obj["bayesian_acmg"] = get_acmg_temperature(
+                        [
+                            criterium["term"]
+                            for criterium in variant_acmg_classifications[0].get("criteria")
```
Comment (northwestwitch) [inline scout/server/blueprints/cases/controllers.py:678]: Collect the latest classification only

File: scout/server/blueprints/cases/controllers.py [hunk 2]
```diff
@@ -663,6 +665,21 @@ def _append_evaluated_variant_by_type(
...
+            ):  # Append info to display the ACMG VUS Bayesian score / temperature
+                variant_acmg_classifications = store.get_evaluations_case_specific(
+                    document_id=var_obj["_id"]
+                )
+                if variant_acmg_classifications:
+                    var_obj["bayesian_acmg"] = get_acmg_temperature(
+                        set(
+                            [
+                                criterium["term"]
+                                for criterium in variant_acmg_classifications[0].get("criteria")
```
Comment (dnil) [inline scout/server/blueprints/cases/controllers.py:679]: Nice! I will note that this seems to fail for some variants/classifications, but I have not figured it out yet. It's not about just having a single-click evaluation, this one had terms.
Screenshot 2024-12-12 at 14 17 02

File: scout/server/blueprints/cases/templates/cases/case_report.html [hunk 3]
```diff
@@ -718,6 +719,13 @@ <h4>Scout case analysis report</h4> - created on:&nbsp;<strong>{{report_created_
               {% else %}
                 -
               {% endif %}
+            <td>
+              {% if variant.bayesian_acmg %}
+                <span class="badge rounded-pill bg-{{variant.bayesian_acmg.temperature_class}}">Score {{variant.bayesian_acmg.points|string}} <span class='fa {{variant.bayesian_acmg.temperature_icon}}'></span> {{variant.bayesian_acmg.temperature}} ({{variant.bayesian_acmg.point_classification}})</span>
```
Comment (dnil) [inline scout/server/blueprints/cases/templates/cases/case_report.html:724]: 💯 Looks good!

File: scout/server/blueprints/cases/templates/cases/case_report.html [hunk 4]
```diff
@@ -718,6 +719,13 @@ <h4>Scout case analysis report</h4> - created on:&nbsp;<strong>{{report_created_
               {% else %}
                 -
               {% endif %}
+            <td>
+              {% if variant.bayesian_acmg %}
```
Comment (dnil) [inline scout/server/blueprints/cases/templates/cases/case_report.html:723]: I guess this test that fails sometimes, but, yeah.

**Review by dnil — APPROVED** [review]
Looks good mostly. I saw an irregular behaviour where the classification would occasionally be missing, so maybe worth spending some more minutes on, but works most of the time. 😸

File: scout/server/blueprints/cases/controllers.py (hunk 2 above)
Comment (northwestwitch) [reply scout/server/blueprints/cases/controllers.py:679]: Those with manual classification (without criteria, I guess?)

Comment (dnil) [reply scout/server/blueprints/cases/controllers.py:679]: Mm, it is an "old" test case, and a busy variant. It doesn't have a current manual, but probably has the events for it from before. Could be some combo of that or deleted classifications? It shows ok on the ACMG classify page so a little annoying. We could test a little on stage perhaps, or on a fresh setup locally?

Comment (northwestwitch) [reply scout/server/blueprints/cases/controllers.py:679]: Which case is that. do you still have the link?

Comment (dnil) [reply scout/server/blueprints/cases/controllers.py:679]: It’s on my local so I can check things against it. Did you deploy to stage yet, otw I can do that and we check a few!

Comment (dnil) [reply scout/server/blueprints/cases/controllers.py:679]: Here is the first variant I tried on stage: 
https://scout-stage.scilifelab.se/evaluations/670ceb0507f2ba4f08471fa6
Does not show temp on 
Screenshot 2024-12-12 at 15 18 10

Comment (northwestwitch) [reply scout/server/blueprints/cases/controllers.py:679]: > It’s on my local so I can check things against it. Did you deploy to stage yet, otw I can do that and we check a few!

I did deploy another branch after this one I think

Comment (northwestwitch) [reply scout/server/blueprints/cases/controllers.py:679]: Ah never mind, you deployed it already

Comment (dnil) [reply scout/server/blueprints/cases/controllers.py:679]: Yes, I've redeployed this one now! Here is another case with the same behaviour:

https://scout-stage.scilifelab.se/cust002/F0007523

Comment (northwestwitch) [reply scout/server/blueprints/cases/controllers.py:679]: That variant actually has criteria:

image

Comment (northwestwitch) [reply scout/server/blueprints/cases/controllers.py:679]: The weird thing is that if I classify a local variant the same way, the it works:

image

image

Comment (dnil) [reply scout/server/blueprints/cases/controllers.py:679]: Yeah, as is evident from the evaluation page with lit up criteria! Same with the others. That evaluation is somewhat recent, but I haven't checked if it also has some other old event somewhere.

Comment (northwestwitch) [reply scout/server/blueprints/cases/controllers.py:679]: The weird thing is that if I classify a local variant the same way then it works

image


image

Comment (dnil) [reply scout/server/blueprints/cases/controllers.py:679]: Concur, and I have not been able to figure out the difference between the ones that fail and the ones that work. I guess both the chain of event-checks to acmg lookup and the `[0]` sorting assumption may be guilty (though the variant above has a date, so something else would have to sneak in). We could add a debug output..

File: scout/utils/acmg.py [hunk 5]
```diff
@@ -303,6 +305,7 @@ def get_acmg_temperature(acmg_terms: set) -> Optional[dict]:
         (points, temperature, point_classification)
 
     """
+    LOG.warning(f"acmg_terms received:{acmg_terms}")
```
Comment (northwestwitch) [inline scout/utils/acmg.py:308]: The right terms arrive to this function:

image

File: scout/server/blueprints/cases/controllers.py (hunk 2 above)
Comment (northwestwitch) [reply scout/server/blueprints/cases/controllers.py:679]: I start to fear that it's a variant re-decoration of sort

File: scout/server/blueprints/cases/controllers.py [hunk 6]
```diff
@@ -680,6 +680,7 @@ def _append_evaluated_variant_by_type(
                             ]
                         )
                     )
+                    print(f"var[bayesian_acmg] is -->{var_obj['bayesian_acmg']}")
```
Comment (northwestwitch) [inline scout/server/blueprints/cases/controllers.py:683]: The right bayesian classification is returned for [this variant](https://scout-stage.scilifelab.se/cust002/F0017133/f40feec9c34fa49398910972e3c1ae8c):

image

File: scout/server/blueprints/cases/controllers.py [hunk 7]
```diff
@@ -624,6 +639,7 @@ def case_report_variants(store: MongoAdapter, case_obj: dict, institute_obj: dic
                 continue
             if case_key == "partial_causatives":
                 var_obj["phenotypes"] = case_obj["partial_causatives"][var_id]
+            add_bayesian_acmg_classification(var_obj)
```
Comment (northwestwitch) [inline scout/server/blueprints/cases/controllers.py:642]: @dnil this is the actual fix: notice that the variants that didn't show the bayesian classif were pinned or causative

File: scout/server/blueprints/cases/controllers.py [hunk 8]
```diff
@@ -663,6 +679,9 @@ def _append_evaluated_variant_by_type(
     """
     for eval_category, variant_key in CASE_REPORT_VARIANT_TYPES.items():
         if variant_key in var_obj and var_obj[variant_key] is not None:
+
+            add_bayesian_acmg_classification(var_obj)
```
Comment (northwestwitch) [inline scout/server/blueprints/cases/controllers.py:683]: This are the other types of variants, like commented, classified, etc

File: scout/server/blueprints/cases/controllers.py (hunk 7 above)
Comment (dnil) [reply scout/server/blueprints/cases/controllers.py:644]: Ah! Well spotted!

File: scout/server/blueprints/cases/controllers.py (hunk 8 above)
Comment (northwestwitch) [reply scout/server/blueprints/cases/controllers.py:685]: I've realised that ACMG classif is shown anyway for all classified variants, that's why this function is always called now

Comment (dnil) [reply scout/server/blueprints/cases/controllers.py:685]: Yep! 👍

**Pull request merged**